File: numberguesser.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *argv[])
{
    int number=1;
    int input=0;
    int score=0;

    if(argc==1) 
    {
        number=100;
    }
    if(argc==2) 
    {
        number=atoi(argv[1]);
    }
    if(argc>=3) 
    {
        exit(1);
    }

    srand(time(0));
    int target = ((rand() % number ) + 1);

        do 
        {
            printf("Enter a number between 1 and %d (inclusive)\n: ", number);
            scanf("%d", &input);  
            score++;
            if (input < target)
            {
                printf("You need to guess higher\n");
            }
            else if (input > target)
            {
                printf("You need to guess lower\n");
            }
        } while (input != target);

    printf("Correct the answer is: %d, you took %d attempt(s)\n", target, score);

    return 0;
}

File: numberguesser.py
#!/usr/bin/env python
import random
import sys

number=1
userInput=0
score=0

if (len(sys.argv)-1) == 0:
  number = 100
else: 
  number = int(sys.argv[1])

target = random.randint(1, number)

while userInput != target:
  print("Enter a number between 1 and %d (inclusive)" % (number))

  userInput = int(input(': '))

  score = score + 1

  if userInput < target:
    print('You need to guess higher')

  if userInput > target:
    print('You need to guess lower')

  if userInput == target:
    break

print("Correct the answer is: %d, you took %d attempt(s)" % (target,score))
